sql/builtin_functions.cc

```diff
@@ -2432,6 +2433,50 @@ BoundFunction flatten(const std::vector<BoundSqlExpression> & args)
 
 static RegisterBuiltin registerFlatten(flatten, "flatten");
 
+BoundFunction clamp(const std::vector<BoundSqlExpression> & args)
+{
+    // Return the result indexed on a single dimension
+
+    checkArgsSize(args.size(), 3);
+
+    return {[=] (const std::vector<ExpressionValue> & args,
+                 const SqlRowScope & scope) -> ExpressionValue
+            {
+                double lower = args[1].toDouble();
+                double upper = args[2].toDouble();
+
+                Date limitsTs = std::max(args[1].getEffectiveTimestamp(), args[2].getEffectiveTimestamp());
+
+                if (args[0].isAtom()) {
+                    double clamped = boost::algorithm::clamp(args[0].toDouble(), lower, upper);
```

Comment:
There is a more pervasive problem... in that 64 bit integer values will end up being molested by being converted to doubles.  Instead, I would define clamp using operator < on the ExpressionValue objects and ignore doubles altogether.